rabbit-source/src/main/java/org/springframework/cloud/stream/module/rabbit/source/RabbitSource.java

```diff
@@ -0,0 +1,126 @@
+import org.springframework.cloud.stream.annotation.Bindings;
+import org.springframework.cloud.stream.annotation.EnableBinding;
+import org.springframework.cloud.stream.messaging.Source;
+import org.springframework.context.annotation.Bean;
+import org.springframework.integration.amqp.inbound.AmqpInboundChannelAdapter;
+import org.springframework.integration.amqp.support.DefaultAmqpHeaderMapper;
+import org.springframework.retry.interceptor.RetryOperationsInterceptor;
+import org.springframework.util.Assert;
+import org.springframework.util.StringUtils;
+
+/**
+ * A source module that receives data from RabbitMQ.
+ *
+ * @author Gary Russell
+ */
+@EnableBinding(Source.class)
+@EnableConfigurationProperties(RabbitSourceProperties.class)
+public class RabbitSource {
+
+	@Autowired
+	@Bindings(RabbitSource.class)
+	private Source channels;
+
+	@Autowired
+	private RabbitProperties rabbitProperties;
+
+	@Autowired
+	private RabbitSourceProperties properties;
+
+	@Autowired(required = false)
+	private ConnectionFactory rabbitConnectionFactory;
+
+	@Bean
+	public SimpleMessageListenerContainer container() {
+		SimpleMessageListenerContainer container = new SimpleMessageListenerContainer(this.rabbitConnectionFactory);
+		container.setAutoStartup(false);
+		Listener listenerProperties = this.rabbitProperties.getListener();
+		AcknowledgeMode acknowledgeMode = listenerProperties.getAcknowledgeMode();
+		if (acknowledgeMode != null) {
+			container.setAcknowledgeMode(acknowledgeMode);
+		}
+		Integer concurrency = listenerProperties.getConcurrency();
+		if (concurrency != null) {
+			container.setConcurrentConsumers(concurrency);
+		}
+		Integer maxConcurrency = listenerProperties.getMaxConcurrency();
+		if (maxConcurrency != null) {
+			container.setMaxConcurrentConsumers(maxConcurrency);
+		}
+		Integer prefetch = listenerProperties.getPrefetch();
+		if (prefetch != null) {
+			container.setPrefetchCount(prefetch);
+		}
+		Integer transactionSize = listenerProperties.getTransactionSize();
+		if (transactionSize != null) {
+			container.setTxSize(transactionSize);
+		}
+		Boolean defaultRequeueRejected = this.properties.getRequeue();
+		if (defaultRequeueRejected != null) {
+			container.setDefaultRequeueRejected(defaultRequeueRejected);
+		}
+		Boolean channelTransacted = this.properties.getChannelTransacted();
+		if (channelTransacted != null) {
+			container.setChannelTransacted(channelTransacted);
+		}
+		String queues = this.properties.getQueues();
+		Assert.state(StringUtils.hasText(queues), "At least one queue is required");
+		container.setQueueNames(StringUtils.commaDelimitedListToStringArray(queues));
+		if (this.properties.isEnableRetry()) {
+			container.setAdviceChain(new Advice[] { rabbitSourceRetryInterceptor() });
+		}
+		return container;
+	}
+
+	@Bean
+	public AmqpInboundChannelAdapter adapter() {
+		AmqpInboundChannelAdapter adapter = new AmqpInboundChannelAdapter(container());
+		adapter.setAutoStartup(false);
```

Comment:
Is this still necessary?
As well as the same for the `container()` above:
```
this.messageListenerContainer = listenerContainer;
this.messageListenerContainer.setAutoStartup(false);
```
M-m-m. Minor, of course, if there is someone who cares about all these our `Lifecycle`s...